src/tests/hotreload-integration-test.ts

```diff
@@ -82,6 +88,85 @@ describe('Hot Code Reload for JS Particle', async () => {
     assert.deepEqual(slotConsumer.getRendering().model,  {name: 'Jack', age: '15'});
     assert.deepEqual(slotConsumer._content.template, `Hello <span>{{name}}</span>, new age: <span>{{age}}</span>`);
   });
+
+  it('ensures new handles are working', async () => {
+    const context = await Manifest.parse(`
+      schema Person
+        Text name
+        Number age
+
+      particle A in 'A.js'
+        in Person personIn
+        out Person personOut
+
+      recipe
+        use as personIn
+        use as personOut
+        A
+          personIn = personIn
+          personOut = personOut
+    `);
+
+    const loader = new StubLoader({
+      'A.js': `defineParticle(({Particle}) => {
+        return class extends Particle {
+          async setHandles(handles) {
+            this.handles = handles;
```

Comment:
Nit: no need to set this; it's not used anywhere else.

Reply:
The reload function in PEC depends on the `handles` field to get all the handles of the old particle and create new handles based on them.

Reply:
So that's a minor problem. We can't assume particles will have set this field, and if we need to rely on it we should set it ourselves. That can wait for another CL.

Reply:
Thanks for noticing, I filed https://github.com/PolymerLabs/arcs/issues/3560 and suggested @sherry-pra talk with you further on approach.

Reply:
I was wondering maybe for this we can either:
1. Set the handles field in `callSetHandles()` function in `particle.ts` so it's always called
2. Create new function maybe `get handles()` or something similar in which different particles can define depending on how they store their handles

Also open to other suggestions :D

Reply:
That is a possible approach. I think we may want to think about the API a bit.